src/processor/operator/persistent/reader/rdf/rdf_reader.cpp

```diff
@@ -76,10 +70,75 @@ static bool supportSerdType(SerdType type) {
     }
 }
 
-static void addSerdNodeToVector(
-    common::ValueVector* vector, uint32_t vectorPos, const SerdNode* node) {
+static void addResourceToVector(
+    common::ValueVector* vector, uint32_t vectorPos, const SerdNode* resourceNode) {
     StringVector::addString(
-        vector, vectorPos, reinterpret_cast<const char*>(node->buf), node->n_bytes);
+        vector, vectorPos, reinterpret_cast<const char*>(resourceNode->buf), resourceNode->n_bytes);
+}
+
+template<typename T>
+static void addVariant(ValueVector* vector, uint32_t pos, LogicalTypeID typeID, T val) {
+    auto typeVector = StructVector::getFieldVector(vector, 0).get();
+    auto valVector = StructVector::getFieldVector(vector, 1).get();
+    typeVector->setValue<uint8_t>(pos, static_cast<uint8_t>(typeID));
+    StringVector::addString(valVector, pos, reinterpret_cast<const char*>(&val), sizeof(T));
+}
+static void addVariant(ValueVector* vector, uint32_t pos, const char* src, uint32_t length) {
+    auto typeVector = StructVector::getFieldVector(vector, 0).get();
+    auto valVector = StructVector::getFieldVector(vector, 1).get();
+    typeVector->setValue<uint8_t>(pos, static_cast<uint8_t>(LogicalTypeID::STRING));
+    StringVector::addString(valVector, pos, reinterpret_cast<const char*>(src), length);
+}
+
+static void addLiteralToVector(common::ValueVector* vector, uint32_t pos,
+    const SerdNode* literalNode, const SerdNode* typeNode) {
+    auto data = reinterpret_cast<const char*>(literalNode->buf);
+    auto length = literalNode->n_bytes;
+    if (typeNode == nullptr) { // No type information available. Treat as string.
+        addVariant(vector, pos, data, length);
+        return;
+    }
+    auto typeData = reinterpret_cast<const char*>(typeNode->buf);
+    auto typeLength = typeNode->n_bytes;
+    bool resolveAsString = true;
+    if (memcmp(XSD.data(), typeData, XSD.size()) == 0) { // Match XSD prefix.
```

Comment:
Can you create a string view over typeData? That avoids the use of `memcmp` and should make the code more intuitive.